src/meta.rs

```diff
@@ -34,32 +38,150 @@ impl ResourceMeta {
         let id = ResourceId::compute(size, &path);
         let name = convert_str(path.file_name());
         let extension = convert_str(path.extension());
-        let modified = metadata.modified()?;
+        let modified = metadata.modified()?.into();
 
-        //todo
-        let kind = None;
-        let extra = None;
+        let kind = Some(ResourceKind::from(path.clone()));
 
         let meta = ResourceMeta {
             id,
             modified,
             name,
             extension,
             kind,
-            extra,
         };
 
         Ok((path.clone(), meta))
     }
 }
 
-//todo
-pub type ResourceKind = ();
-pub type ResourceExtra = ();
+#[derive(Eq, PartialEq, Hash, Clone, Debug, EnumString, Display, EnumCount)]
+#[strum(ascii_case_insensitive)]
+pub enum ResourceKind {
```

Comment:
I think we should port resource kinds in the last turn... the same as previews. First, just plain interaction with Android should be set up.

Reply:
We can just pass `None` before we implement resource kinds.
```
val kind: ResourceKind?
```

Reply:
Agree.